src/libecalc/presentation/exporter/queries.py

```diff
@@ -96,28 +97,30 @@ def query(
             for fuel_consumer in installation_dto.fuel_consumers:
                 temporal_category = TemporalModel(fuel_consumer.user_defined_category)
                 for period, category in temporal_category.items():
-                    if self.consumer_categories is None or category in self.consumer_categories:
-                        fuel_consumer_result = installation_graph.get_energy_result(fuel_consumer.id)
-                        fuel_volumes = (
-                            TimeSeriesRate.from_timeseries_stream_day_rate(
-                                fuel_consumer_result.energy_usage, regularity=regularity
+                    if Period.intersects(period, installation_graph.variables_map.period):
+                        if self.consumer_categories is None or category in self.consumer_categories:
+                            period = Period.intersection(period, installation_graph.variables_map.period)
+                            fuel_consumer_result = installation_graph.get_energy_result(fuel_consumer.id)
+                            fuel_volumes = (
+                                TimeSeriesRate.from_timeseries_stream_day_rate(
+                                    fuel_consumer_result.energy_usage, regularity=regularity
+                                )
+                                .for_period(period)
+                                .to_volumes_for_period(period)
```

Comment:
Not sure why we need both `for_period` and `to_volumes_for_period`. Either `to_volumes_for_period` or `for_period(period).to_volumes()` makes sense to me.